File: src/java/magma/api/collect/RangeHead.java
package magma.api.collect;

import magma.api.option.Option;
import magma.app.Main;

public final class RangeHead implements Head<Integer> {
    private final int length;
    private int counter;

    public RangeHead(int length) {
        this.length = length;
        this.counter = 0;
    }

    @Override
    public Option<Integer> next() {
        if (this.counter >= this.length) {
            return new Main.None<Integer>();
        }

        var value = this.counter;
        this.counter++;
        return new Main.Some<Integer>(value);
    }
}

File: src/java/magma/api/collect/RangeHead.ts
import { Option } from "../../../magma/api/option/Option";
import { Main } from "../../../magma/app/Main";
export class RangeHead implements Head<number> {
	length: number;
	counter: number;
	constructor (length: number) {
		this.length = length;
		this.counter = 0;
	}
	next(): Option<number> {
		if (this.counter >= this.length){
			return new Main.None<number>();
		}
		let value = this.counter;
		this.counter++;
		return new Main.Some<number>(value);
	}
}
